### DISCUSSION TITLE
How to open the datepicker programatically?

### DISCUSSION THREAD

**Discussion by aspyropoulos** [ankurk91/vue-flatpickr-component#242] [Q&A] [answered]
The [flatpickr](https://flatpickr.js.org/instance-methods-properties-elements/) docs have the ability to open the datepicker programmatically by using the open() function ie

```
let calendar = flatpickr(yourElement, config);
calendar.open()
```

How can I do this with your plugin in vuejs?

**Comment by aspyropoulos** [comment, marked as answer]
I just found out. I used the below expression:

`<flat-pickr ref="datepicker" v-model="date"></flat-pickr>`

```
    methods: {
        open: function() {
            this.$refs.datepicker.fp.open();
        }
    },
```

**Comment by ankurk91** [comment]
https://flatpickr.js.org/instance-methods-properties-elements/
